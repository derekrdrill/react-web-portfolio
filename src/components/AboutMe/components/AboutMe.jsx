import React from 'react';
import styled, { createGlobalStyle } from 'styled-components';
import { Container, Row, Col } from 'react-bootstrap';
import { Typography, List, ListItem } from '@mui/material';

import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import { ABOUT_ME_TEXT } from '../constants/ABOUT_ME_TEXT';
import bitmojiChillinWithBirds from '../../../assets/bitmoji_chillinWithBirds1.png';

export const AboutMe = ({ id }) => {
  // test change

  const { darkMode } = React.useContext(DarkLightModeContext);

  return (
    <>
      <PageBodyStyle darkMode={darkMode} />
      <AboutMeContainer darkMode={darkMode} id={id} fluid>
        <Row>
          <Col className='py-3'>
            <AboutMeTitle className='text-center' component='h1' darkMode={darkMode} variant='h3'>
              ABOUT ME
            </AboutMeTitle>
          </Col>
        </Row>
        <Row>
          <AboutMeTextContainer
            darkMode={darkMode}
            xs={{ span: 10, order: 'last', offset: 1 }}
            lg={{ span: 5, order: 'first', offset: 1 }}
          >
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[1].text}
            </AboutMeText>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[2].text}
            </AboutMeText>
            <List style={{ color: 'white' }}>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[3].text}</AboutMeListItem>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[4].text}</AboutMeListItem>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[5].text}</AboutMeListItem>
              <AboutMeListItem darkMode={darkMode}>{ABOUT_ME_TEXT[6].text}</AboutMeListItem>
            </List>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[7].text}
            </AboutMeText>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[8].text}
            </AboutMeText>
            <AboutMeText darkMode={darkMode} variant='subtitle1' component='p'>
              {ABOUT_ME_TEXT[9].text}
            </AboutMeText>
          </AboutMeTextContainer>
          <AboutMeBitmojiContainer xs={12} lg={6}>
            <img src={bitmojiChillinWithBirds} alt='Chillin with birds' />
          </AboutMeBitmojiContainer>
        </Row>
      </AboutMeContainer>
    </>
  );
};

const PageBodyStyle = createGlobalStyle(({ darkMode }) => ({
  body: {
    backgroundColor: darkMode ? '#030200' : 'whitesmoke',
  },
}));

export const AboutMeContainer = styled(Container)(({ darkMode }) => ({
  padding: '30px 50px 200px 50px',
  borderTop: '2px solid transparent',
  borderImage: darkMode
    ? 'linear-gradient(to right, rgba(248, 184, 255, 1), skyblue, gainsboro)'
    : 'linear-gradient(to right, rgba(176, 52, 197, 1), rgba(86, 206, 210, 1))',
  borderImageSlice: 1,
}));

export const AboutMeTitle = styled(Typography)(({ darkMode }) => ({
  fontFamily: 'Shizuru',
  fontWeight: darkMode ? 'normal' : 'bold',
  color: darkMode ? 'gainsboro' : 'darkslategray',
}));

export const AboutMeTextContainer = styled(Col)(({ darkMode }) => ({
  padding: 40,
  height: 550,
  overflowY: 'auto',
  border: '1px dashed #383838',
  borderRadius: 2,
  backgroundColor: darkMode ? '#303030' : '#eee5c1',
}));

export const AboutMeBitmojiContainer = styled(Col)({
  textAlign: 'center',
  paddingTop: 75,
  paddingBottom: 20,
});

export const AboutMeText = styled(Typography)(({ darkMode }) => ({
  fontFamily: 'Kufam',
  color: darkMode ? 'gainsboro' : '#333333',
  marginBottom: 15,
}));

export const AboutMeListItem = styled(ListItem)(({ darkMode }) => ({
  fontFamily: 'Kufam',
  color: darkMode ? 'gainsboro' : '#333333',
}));
