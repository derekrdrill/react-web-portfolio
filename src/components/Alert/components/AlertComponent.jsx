import React, { useContext } from 'react';
import styled, { css, keyframes } from 'styled-components';
import { Alert, AlertTitle } from '@mui/material';
import { AlertContext } from '../context/AlertContext';

export const AlertComponent = () => {
  const { alert, fadeOut } = useContext(AlertContext);

  return (
    alert && (
      <StyledAlert severity={alert.type} $fadeOut={fadeOut}>
        <AlertTitle>{alert.title}</AlertTitle>
        {alert.msg}
      </StyledAlert>
    )
  );
};

const fadeOut = keyframes({
  '0%': {
    opacity: 1,
  },
  '100%': {
    opacity: 0,
  },
});

const StyledAlert = styled(Alert)`
  animation-name: ${props =>
    props.$fadeOut
      ? css`
          ${fadeOut}
        `
      : 'none'};
  animation-duration: 500ms;
  animation-direction: linear;
`;
