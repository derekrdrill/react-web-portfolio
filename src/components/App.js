import React from 'react';

const AboutMe = React.lazy(() => import('./AboutMe/components/AboutMe'));
const ConnectWithMe = React.lazy(() => import('./ConnectWithMe/components/ConnectWithMe'));
const Homepage = React.lazy(() => import('./Homepage/Homepage'));
const Projects = React.lazy(() => import('./Projects/components/Projects'));

const App = () => (
  <div className='App'>
    <Homepage id='home' />
    <AboutMe id='about-me' />
    <Projects id='projects' />
    <ConnectWithMe id='connect-with-me' />
  </div>
);

export default App;
