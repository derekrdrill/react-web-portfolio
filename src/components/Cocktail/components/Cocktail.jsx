import React from 'react';
import PropTypes from 'prop-types';
import styled, { createGlobalStyle } from 'styled-components';

import { AlertProvider } from '../../Alert/context/AlertContext';
import { DarkLightModeContext } from '../../DarkLightMode/context/DarkLightModeContext';

import CocktailDetail from './CocktailDetail';
import CocktailSearch from './CocktailSearch';

const Cocktail = () => {
  const { darkMode } = React.useContext(DarkLightModeContext);

  return (
    <AlertProvider>
      <PageStyle darkMode={darkMode} />
      <CocktailSearch />
      <CocktailDetail />
    </AlertProvider>
  );
};

Cocktail.propTypes = {};

export default Cocktail;

const PageStyle = createGlobalStyle(({ darkMode }) => ({
  body: {
    backgroundColor: darkMode ? '#1B0008' : '#FBAF93',
  },
  'h1, h2, h3, h4, h5, h6, p': {
    fontFamily: 'Trebuchet MS, sans-serif',
    fontWeight: 'bold',
    color: darkMode ? 'beige' : '#464646',
  },
}));
