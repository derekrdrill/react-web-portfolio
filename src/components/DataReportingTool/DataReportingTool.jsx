import React, { useState, useEffect } from 'react';
import styled, { createGlobalStyle } from 'styled-components';
import {
  Select,
  InputLabel,
  FormControl,
  Typography,
  MenuItem,
  List,
  ListItem,
  IconButton,
  Grid,
  Button,
} from '@mui/material';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faLongArrowAltDown, faLongArrowAltUp } from '@fortawesome/fontawesome-free-solid';
import AddCircleIcon from '@mui/icons-material/AddCircle';
import RemoveCircleIcon from '@mui/icons-material/RemoveCircle';
import { DynamicDataTable } from '../DynamicDataTable/DynamicDataTable';

const DynamicColumnsList = ({ columns, initialList = false, buttonClick }) => (
  <ColumnsListContainer>
    <ColumnsList>
      {columns.map(column => (
        <ColumnsListItem key={column.id} draggable='true'>
          <Grid container>
            <Grid item xs={10} order={{ xs: 1, lg: initialList ? 1 : 2 }}>
              <Grid container justifyContent={{ xs: 'flex-start', lg: initialList ? 'flex-start' : 'flex-end' }}>
                <Typography variant='subtitle1' component='h3'>
                  {column.columnName || ''}
                </Typography>
              </Grid>
            </Grid>
            <Grid item xs={2} order={{ xs: 2, lg: initialList ? 2 : 1 }}>
              <Grid container justifyContent={{ xs: 'flex-end', lg: initialList ? 'flex-end' : 'flex-start' }}>
                <IconButton
                  id={column.id}
                  color={initialList ? 'success' : 'error'}
                  children={initialList ? <AddCircleIcon /> : <RemoveCircleIcon />}
                  onClick={e => buttonClick(e, initialList)}
                  variant='contained'
                />
              </Grid>
            </Grid>
          </Grid>
        </ColumnsListItem>
      ))}
    </ColumnsList>
  </ColumnsListContainer>
);
const MoveAllColumnsButton = ({ children, endIcon, onClick }) => (
  <Button children={children} color='secondary' endIcon={endIcon} fullWidth onClick={onClick} variant='contained' />
);
export const DataReportingTool = () => {
  const [dataSet, setDataSet] = useState('Select One');
  const [collections, setCollections] = useState([]);
  const [collectionKeys, setCollectionKeys] = useState([]);
  const [availableColumns, setAvailableColumns] = useState([]);
  const [associatedColumns, setAssociatedColumns] = useState([]);
  const [sortedColumns, setSortedColumns] = useState([]);

  const handleGetAllCollections = () => {
    fetch('http://localhost:3001/listCollections')
      .then(response => response.json())
      .then(json => setCollections(json));
  };

  const handleGetCollectionKeys = () => {
    fetch(`http://localhost:3001/listCollectionKeys/${dataSet}`)
      .then(response => response.json())
      .then(json => setCollectionKeys(json));
  };

  const handleColumnState = (e, initialList) => {
    let columnID = Number(e.currentTarget.id);
    let columns = initialList ? availableColumns : associatedColumns;
    let columnObject = columns.find(column => column.id === columnID);

    if (initialList) {
      setAvailableColumns(availableColumns.filter(column => column.id !== columnID));
      setAssociatedColumns([...associatedColumns, columnObject]);
    } else {
      setAssociatedColumns(associatedColumns.filter(column => column.id !== columnID));
      setAvailableColumns([...availableColumns, columnObject]);
    }
  };

  const handleAssociateAll = () => {
    if (availableColumns.length > 0) {
      setAssociatedColumns(associatedColumns.concat(availableColumns));
      setAvailableColumns([]);
    }
  };

  const handleDeassociateAll = () => {
    if (associatedColumns.length > 0) {
      setAvailableColumns(availableColumns.concat(associatedColumns));
      setAssociatedColumns([]);
    }
  };

  useEffect(() => handleGetAllCollections(), []);
  useEffect(() => handleGetCollectionKeys(), [dataSet]);
  useEffect(() => setAvailableColumns(collectionKeys), [collectionKeys]);

  return (
    <div>
      <PageBodyStyle />
      <DataSetSelectContainer container>
        <Grid item xs={3}>
          <FormControl fullWidth>
            <InputLabel>
              <Typography variant='subtitle1' component='label'>
                Select dataset
              </Typography>
            </InputLabel>
            <DataSetSelect
              label={dataSet}
              variant='standard'
              onChange={e => setDataSet(e.target.value)}
              value={dataSet ?? ''}
            >
              {collections.map(collection => (
                <MenuItem key={collection.name} value={collection.name}>
                  {collection.name}
                </MenuItem>
              ))}
              {/* <MenuItem value='leadData'>Lead Data</MenuItem> */}
            </DataSetSelect>
          </FormControl>
        </Grid>
      </DataSetSelectContainer>
      <ColumnsSelectContainer container>
        <Grid item xs={12} lg={5}>
          <Typography variant='h5' component='h2'>
            Available Columns
          </Typography>
          <DynamicColumnsList columns={availableColumns} buttonClick={handleColumnState} initialList />
        </Grid>
        <MoveAllColumnsButtonContainer item xs={12} lg={2}>
          <Grid container justifyContent='center' spacing={{ xs: 1, lg: 2 }}>
            <Grid item xs={6} lg={7} order={{ xs: 2, lg: 1 }}>
              <Grid item display={{ xs: 'none', lg: 'block' }}>
                <MoveAllColumnsButton children='>' onClick={handleAssociateAll} />
              </Grid>
              <Grid item display={{ xs: 'block', lg: 'none' }}>
                <MoveAllColumnsButton
                  children='ASSOCIATE ALL'
                  endIcon={<FontAwesomeIcon icon={faLongArrowAltDown} />}
                  onClick={handleAssociateAll}
                />
              </Grid>
            </Grid>
            <Grid item xs={6} lg={7} order={{ xs: 1, lg: 2 }}>
              <Grid item display={{ xs: 'none', lg: 'block' }}>
                <MoveAllColumnsButton children='<' onClick={handleDeassociateAll} />
              </Grid>
              <Grid item display={{ xs: 'block', lg: 'none' }}>
                <MoveAllColumnsButton
                  children='DEASSOCIATE ALL'
                  endIcon={<FontAwesomeIcon icon={faLongArrowAltUp} />}
                  onClick={handleDeassociateAll}
                />
              </Grid>
            </Grid>
          </Grid>
        </MoveAllColumnsButtonContainer>
        <Grid item xs={12} lg={5}>
          <Typography variant='h5' component='h2'>
            Associated Columns
          </Typography>
          <DynamicColumnsList columns={associatedColumns} buttonClick={handleColumnState} />
        </Grid>
      </ColumnsSelectContainer>
      <hr />
      <Grid container justifyContent='center'>
        <Grid item xs={12} lg={4}>
          <Button color='info' variant='contained' children={'Create Report'} fullWidth />
        </Grid>
      </Grid>
    </div>
  );
};

const PageBodyStyle = createGlobalStyle({
  body: {
    backgroundColor: '#228881',
  },
});

const DataSetSelectContainer = styled(Grid)({
  backgroundColor: 'grey',
  position: '-webkit-sticky',
  position: 'sticky',
  top: 80,
  padding: '10px 0 15px 0',
  zIndex: 100,
});

const DataSetSelect = styled(Select)({
  backgroundColor: 'gainsboro',
  paddingLeft: 10,
  borderRadius: 5,
  marginLeft: 10,
});

const ColumnsSelectContainer = styled(Grid)({
  padding: 20,
});

const ColumnsListContainer = styled.div({
  height: 250,
  overflowY: 'auto',
  overflowX: 'none',
  border: '1px solid grey',
  borderRadius: 5,
  boxShadow: '0 0 3px darkslategray',
  backgroundColor: '#EDE9E1',
});

const ColumnsList = styled(List)({
  padding: 0,
});

const ColumnsListItem = styled(ListItem)({
  backgroundColor: 'darkgrey',
  borderBottom: '1px dashed grey',
  cursor: 'all-scroll',
  padding: 10,
});

const MoveAllColumnsButtonContainer = styled(Grid)({
  margin: '7% 0 4% 0',
});
