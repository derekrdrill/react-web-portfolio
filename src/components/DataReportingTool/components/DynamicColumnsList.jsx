import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Grid, IconButton, List, ListItem, Typography } from '@mui/material';
import AddCircleIcon from '@mui/icons-material/AddCircle';
import RemoveCircleIcon from '@mui/icons-material/RemoveCircle';

export const getColumnName = columnName => columnName ?? '';
export const getIconButtonColor = initialList => (initialList ? 'success' : 'error');
export const getIconButtonChildren = initialList => (initialList ? <AddCircleIcon /> : <RemoveCircleIcon />);
export const getFirstGridItemOrderLG = initialList => (initialList ? 1 : 2);
export const getSecondGridItemOrderLG = initialList => (initialList ? 2 : 1);
export const getFirstGridContainerJustifyLG = initialList => (initialList ? 'flex-start' : 'flex-end');
export const getSecondGridContainerJustifyLG = initialList => (initialList ? 'flex-end' : 'flex-start');

export const DynamicColumnsList = ({ columns, initialList = false, buttonClick }) => (
  <ColumnsListContainer>
    <ColumnsList>
      {columns.map(column => (
        <ColumnsListItem key={column.id} draggable='true'>
          <Grid container>
            <Grid item xs={10} order={{ xs: 1, lg: getFirstGridItemOrderLG(initialList) }}>
              <Grid
                container
                justifyContent={{
                  xs: 'flex-start',
                  lg: getFirstGridContainerJustifyLG(initialList),
                }}
              >
                <Typography variant='subtitle1' component='h3'>
                  {getColumnName(column.columnName)}
                </Typography>
              </Grid>
            </Grid>
            <Grid item xs={2} order={{ xs: 2, lg: getSecondGridItemOrderLG(initialList) }}>
              <Grid
                container
                justifyContent={{
                  xs: 'flex-end',
                  lg: getSecondGridContainerJustifyLG(initialList),
                }}
              >
                <IconButton
                  id={column.id}
                  color={getIconButtonColor(initialList)}
                  onClick={
                    /* istanbul ignore next */
                    e => buttonClick(e, initialList)
                  }
                  variant='contained'
                >
                  {getIconButtonChildren(initialList)}
                </IconButton>
              </Grid>
            </Grid>
          </Grid>
        </ColumnsListItem>
      ))}
    </ColumnsList>
  </ColumnsListContainer>
);

DynamicColumnsList.propTypes = {};

const ColumnsListContainer = styled.div({
  height: 250,
  overflowY: 'auto',
  overflowX: 'none',
  border: '1px solid grey',
  borderRadius: 5,
  boxShadow: '0 0 3px darkslategray',
  backgroundColor: '#EDE9E1',
});

const ColumnsList = styled(List)({
  padding: 0,
});

const ColumnsListItem = styled(ListItem)({
  backgroundColor: 'darkgrey',
  borderBottom: '1px dashed grey',
  cursor: 'all-scroll',
  padding: 10,
});
