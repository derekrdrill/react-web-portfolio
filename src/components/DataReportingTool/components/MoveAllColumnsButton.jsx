import React from 'react';
import PropTypes from 'prop-types';
import { Button } from '@mui/material';

export const MoveAllColumnsButton = ({ children, endIcon, onClick }) => (
  <Button children={children} color='secondary' endIcon={endIcon} fullWidth onClick={onClick} variant='contained' />
);
