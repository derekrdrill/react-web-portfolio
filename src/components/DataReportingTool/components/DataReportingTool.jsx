import React, { useState, useEffect } from 'react';
import styled, { createGlobalStyle } from 'styled-components';
import { Select, InputLabel, FormControl, Typography, MenuItem, Grid, Button } from '@mui/material';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faLongArrowAltDown, faLongArrowAltUp } from '@fortawesome/fontawesome-free-solid';

import { DynamicColumnsList } from '../components/DynamicColumnsList';
import { MoveAllColumnsButton } from '../components/MoveAllColumnsButton';

export const DataReportingTool = () => {
  const [dataSet, setDataSet] = useState('Select One');
  const [collections, setCollections] = useState([]);
  const [collectionKeys, setCollectionKeys] = useState([]);
  const [availableColumns, setAvailableColumns] = useState([]);
  const [associatedColumns, setAssociatedColumns] = useState([]);

  const handleGetAllCollections = () => {
    fetch('http://localhost:3001/listCollections')
      .then(response => response.json())
      .then(json => setCollections(json));
  };

  const handleGetCollectionKeys = React.useCallback(() => {
    fetch(`http://localhost:3001/listCollectionKeys/${dataSet}`)
      .then(response => response.json())
      .then(json => setCollectionKeys(json));
  }, [dataSet]);

  const handleColumnState = (e, initialList) => {
    let columnID = Number(e.currentTarget.id);
    let columns = initialList ? availableColumns : associatedColumns;
    let columnObject = columns.find(column => column.id === columnID);

    if (initialList) {
      setAvailableColumns(availableColumns.filter(column => column.id !== columnID));
      setAssociatedColumns([...associatedColumns, columnObject]);
    } else {
      setAssociatedColumns(associatedColumns.filter(column => column.id !== columnID));
      setAvailableColumns([...availableColumns, columnObject]);
    }
  };

  const handleAssociateAll = () => {
    if (availableColumns.length > 0) {
      setAssociatedColumns(associatedColumns.concat(availableColumns));
      setAvailableColumns([]);
    }
  };

  const handleDeassociateAll = () => {
    if (associatedColumns.length > 0) {
      setAvailableColumns(availableColumns.concat(associatedColumns));
      setAssociatedColumns([]);
    }
  };

  useEffect(() => handleGetAllCollections(), []);
  useEffect(() => handleGetCollectionKeys(), [dataSet, handleGetCollectionKeys]);
  useEffect(() => setAvailableColumns(collectionKeys), [collectionKeys]);

  return (
    <div>
      <PageBodyStyle />
      <DataSetSelectContainer container>
        <Grid item xs={3}>
          <FormControl fullWidth>
            <InputLabel>
              <Typography variant='subtitle1' component='label'>
                Select dataset
              </Typography>
            </InputLabel>
            <DataSetSelect
              label={dataSet}
              variant='standard'
              onChange={e => setDataSet(e.target.value)}
              value={dataSet ?? ''}
            >
              {collections.map(collection => (
                <MenuItem key={collection.name} value={collection.name}>
                  {collection.name}
                </MenuItem>
              ))}
              {/* <MenuItem value='leadData'>Lead Data</MenuItem> */}
            </DataSetSelect>
          </FormControl>
        </Grid>
      </DataSetSelectContainer>
      <ColumnsSelectContainer container>
        <Grid item xs={12} lg={5}>
          <Typography variant='h5' component='h2'>
            Available Columns
          </Typography>
          <DynamicColumnsList
            columns={availableColumns}
            buttonClick={handleColumnState}
            initialList
          />
        </Grid>
        <MoveAllColumnsButtonContainer item xs={12} lg={2}>
          <Grid container justifyContent='center' spacing={{ xs: 1, lg: 2 }}>
            <Grid item xs={6} lg={7} order={{ xs: 2, lg: 1 }}>
              <Grid item display={{ xs: 'none', lg: 'block' }}>
                <MoveAllColumnsButton onClick={handleAssociateAll}>{`>`}</MoveAllColumnsButton>
              </Grid>
              <Grid item display={{ xs: 'block', lg: 'none' }}>
                <MoveAllColumnsButton
                  endIcon={<FontAwesomeIcon icon={faLongArrowAltDown} />}
                  onClick={handleAssociateAll}
                >
                  {`ASSOCIATE ALL`}
                </MoveAllColumnsButton>
              </Grid>
            </Grid>
            <Grid item xs={6} lg={7} order={{ xs: 1, lg: 2 }}>
              <Grid item display={{ xs: 'none', lg: 'block' }}>
                <MoveAllColumnsButton onClick={handleDeassociateAll}>{`<`}</MoveAllColumnsButton>
              </Grid>
              <Grid item display={{ xs: 'block', lg: 'none' }}>
                <MoveAllColumnsButton
                  endIcon={<FontAwesomeIcon icon={faLongArrowAltUp} />}
                  onClick={handleDeassociateAll}
                >
                  {`DEASSOCIATE ALL`}
                </MoveAllColumnsButton>
              </Grid>
            </Grid>
          </Grid>
        </MoveAllColumnsButtonContainer>
        <Grid item xs={12} lg={5}>
          <Typography variant='h5' component='h2'>
            Associated Columns
          </Typography>
          <DynamicColumnsList columns={associatedColumns} buttonClick={handleColumnState} />
        </Grid>
      </ColumnsSelectContainer>
      <hr />
      <Grid container justifyContent='center'>
        <Grid item xs={12} lg={4}>
          <Button color='info' variant='contained' fullWidth>{`Create Report`}</Button>
        </Grid>
      </Grid>
    </div>
  );
};

const PageBodyStyle = createGlobalStyle({
  body: {
    backgroundColor: '#228881',
  },
});

const DataSetSelectContainer = styled(Grid)({
  backgroundColor: 'grey',
  // position: '-webkit-sticky',
  position: 'sticky',
  top: 80,
  padding: '10px 0 15px 0',
  zIndex: 100,
});

const DataSetSelect = styled(Select)({
  backgroundColor: 'gainsboro',
  paddingLeft: 10,
  borderRadius: 5,
  marginLeft: 10,
});

const ColumnsSelectContainer = styled(Grid)({
  padding: 20,
});

// const ColumnsListContainer = styled.div({
//   height: 250,
//   overflowY: 'auto',
//   overflowX: 'none',
//   border: '1px solid grey',
//   borderRadius: 5,
//   boxShadow: '0 0 3px darkslategray',
//   backgroundColor: '#EDE9E1',
// });

// const ColumnsList = styled(List)({
//   padding: 0,
// });

// const ColumnsListItem = styled(ListItem)({
//   backgroundColor: 'darkgrey',
//   borderBottom: '1px dashed grey',
//   cursor: 'all-scroll',
//   padding: 10,
// });

const MoveAllColumnsButtonContainer = styled(Grid)({
  margin: '7% 0 4% 0',
});
