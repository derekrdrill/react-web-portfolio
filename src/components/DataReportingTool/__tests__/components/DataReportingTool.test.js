import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';

import { configure } from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

import { DataReportingTool } from '../../components/DataReportingTool';

configure({ adapter: new Adapter() });

const renderer = new ShallowRenderer();

describe('DataReportingTool with me tests', () => {
  it('renders correctly', () => {
    renderer.render(<DataReportingTool />);
    const result = renderer.getRenderOutput();
    expect(result).toMatchSnapshot();
  });

  //   it('renders DarkLightModeIcon styled component correctly', () => {
  //     const darkLightModeIcon = shallow(<DarkLightModeIcon />);
  //     const darkLightModeIconDark = shallow(<DarkLightModeIcon darkMode />);

  //     expect(darkLightModeIcon.props().className).toEqual('sc-bczRLJ cFhdZs');
  //     expect(darkLightModeIconDark.props().className).toEqual('sc-bczRLJ iBihyU');
  //   });

  //   it('runs getIcon correctly', () => {
  //     expect(getIcon(true, faSun, faMoon)).toEqual(faSun);
  //     expect(getIcon(false, faSun, faMoon)).toEqual(faMoon);
  //   });
});
