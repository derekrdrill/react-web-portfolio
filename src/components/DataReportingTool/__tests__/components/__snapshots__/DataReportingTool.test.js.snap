// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataReportingTool with me tests renders correctly 1`] = `
<div>
  <Memo(l) />
  <Styled(Component)
    container={true}
  >
    <ForwardRef(Grid)
      item={true}
      xs={3}
    >
      <ForwardRef(FormControl)
        fullWidth={true}
      >
        <ForwardRef(InputLabel)>
          <ForwardRef(Typography)
            component="label"
            variant="subtitle1"
          >
            Select dataset
          </ForwardRef(Typography)>
        </ForwardRef(InputLabel)>
        <Styled(Component)
          label="Select One"
          onChange={[Function]}
          value="Select One"
          variant="standard"
        />
      </ForwardRef(FormControl)>
    </ForwardRef(Grid)>
  </Styled(Component)>
  <Styled(Component)
    container={true}
  >
    <ForwardRef(Grid)
      item={true}
      lg={5}
      xs={12}
    >
      <ForwardRef(Typography)
        component="h2"
        variant="h5"
      >
        Available Columns
      </ForwardRef(Typography)>
      <DynamicColumnsList
        buttonClick={[Function]}
        columns={Array []}
        initialList={true}
      />
    </ForwardRef(Grid)>
    <Styled(Component)
      item={true}
      lg={2}
      xs={12}
    >
      <ForwardRef(Grid)
        container={true}
        justifyContent="center"
        spacing={
          Object {
            "lg": 2,
            "xs": 1,
          }
        }
      >
        <ForwardRef(Grid)
          item={true}
          lg={7}
          order={
            Object {
              "lg": 1,
              "xs": 2,
            }
          }
          xs={6}
        >
          <ForwardRef(Grid)
            display={
              Object {
                "lg": "block",
                "xs": "none",
              }
            }
            item={true}
          >
            <MoveAllColumnsButton
              onClick={[Function]}
            >
              &gt;
            </MoveAllColumnsButton>
          </ForwardRef(Grid)>
          <ForwardRef(Grid)
            display={
              Object {
                "lg": "none",
                "xs": "block",
              }
            }
            item={true}
          >
            <MoveAllColumnsButton
              endIcon={
                <FontAwesomeIcon
                  beat={false}
                  beatFade={false}
                  border={false}
                  bounce={false}
                  className=""
                  fade={false}
                  fixedWidth={false}
                  flip={false}
                  icon={
                    Object {
                      "icon": Array [
                        256,
                        512,
                        Array [],
                        "f309",
                        "M168 345.941V44c0-6.627-5.373-12-12-12h-56c-6.627 0-12 5.373-12 12v301.941H41.941c-21.382 0-32.09 25.851-16.971 40.971l86.059 86.059c9.373 9.373 24.569 9.373 33.941 0l86.059-86.059c15.119-15.119 4.411-40.971-16.971-40.971H168z",
                      ],
                      "iconName": "long-arrow-alt-down",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  maskId={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  shake={false}
                  size={null}
                  spin={false}
                  spinPulse={false}
                  spinReverse={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  titleId={null}
                  transform={null}
                />
              }
              onClick={[Function]}
            >
              ASSOCIATE ALL
            </MoveAllColumnsButton>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
        <ForwardRef(Grid)
          item={true}
          lg={7}
          order={
            Object {
              "lg": 2,
              "xs": 1,
            }
          }
          xs={6}
        >
          <ForwardRef(Grid)
            display={
              Object {
                "lg": "block",
                "xs": "none",
              }
            }
            item={true}
          >
            <MoveAllColumnsButton
              onClick={[Function]}
            >
              &lt;
            </MoveAllColumnsButton>
          </ForwardRef(Grid)>
          <ForwardRef(Grid)
            display={
              Object {
                "lg": "none",
                "xs": "block",
              }
            }
            item={true}
          >
            <MoveAllColumnsButton
              endIcon={
                <FontAwesomeIcon
                  beat={false}
                  beatFade={false}
                  border={false}
                  bounce={false}
                  className=""
                  fade={false}
                  fixedWidth={false}
                  flip={false}
                  icon={
                    Object {
                      "icon": Array [
                        256,
                        512,
                        Array [],
                        "f30c",
                        "M88 166.059V468c0 6.627 5.373 12 12 12h56c6.627 0 12-5.373 12-12V166.059h46.059c21.382 0 32.09-25.851 16.971-40.971l-86.059-86.059c-9.373-9.373-24.569-9.373-33.941 0l-86.059 86.059c-15.119 15.119-4.411 40.971 16.971 40.971H88z",
                      ],
                      "iconName": "long-arrow-alt-up",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  maskId={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  shake={false}
                  size={null}
                  spin={false}
                  spinPulse={false}
                  spinReverse={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  titleId={null}
                  transform={null}
                />
              }
              onClick={[Function]}
            >
              DEASSOCIATE ALL
            </MoveAllColumnsButton>
          </ForwardRef(Grid)>
        </ForwardRef(Grid)>
      </ForwardRef(Grid)>
    </Styled(Component)>
    <ForwardRef(Grid)
      item={true}
      lg={5}
      xs={12}
    >
      <ForwardRef(Typography)
        component="h2"
        variant="h5"
      >
        Associated Columns
      </ForwardRef(Typography)>
      <DynamicColumnsList
        buttonClick={[Function]}
        columns={Array []}
      />
    </ForwardRef(Grid)>
  </Styled(Component)>
  <hr />
  <ForwardRef(Grid)
    container={true}
    justifyContent="center"
  >
    <ForwardRef(Grid)
      item={true}
      lg={4}
      xs={12}
    >
      <ForwardRef(Button)
        color="info"
        fullWidth={true}
        variant="contained"
      >
        Create Report
      </ForwardRef(Button)>
    </ForwardRef(Grid)>
  </ForwardRef(Grid)>
</div>
`;
